source/org/zfin/ontology/repository/HibernateOntologyRepository.java

```diff
@@ -34,553 +38,558 @@
+				Set<TermDTO> parentTerms = termDTO.getParentTerms();
+				if (parentTerms == null) {
+					parentTerms = new HashSet<TermDTO>();
+				}
+				parentTerms.add(parentDTO);
+				termDTO.setParentTerms(parentTerms);
+			}
+
+			termDTOMap.put(result[0].toString(), termDTO);
+		}
+
+		return termDTOMap;
+	}
+
+	@Override
+	public int getNumberTermsForOntology(Ontology ontology) {
+		String hql = " " +
+			" select count(*) from GenericTerm t  " +
+			" where t.ontology = :ontology  " +
+			" and t.secondary = :secondary " +
+			"";
+		Session session = HibernateUtil.currentSession();
+		Query query = session.createQuery(hql);
+		query.setParameter("ontology", ontology);
+		query.setBoolean("secondary", false);
+		return Integer.valueOf(query.uniqueResult().toString()).intValue();
+	}
+
+	/**
+	 * Retrieve a map of aliases that match a term of a given ontology.
+	 * An alias has to be unique within an ontology.
+	 *
+	 * @param ontology Ontology
+	 * @return list of TermAlias objects
+	 */
+	@SuppressWarnings("unchecked")
+	public List<TermAlias> getAllAliases(Ontology ontology) {
+		Session session = HibernateUtil.currentSession();
+		String hql = " from TermAlias as alias where " +
+			"          alias.term.obsolete <> :obsolete AND " +
+			"          alias.term.ontology = :ontology ";
+
+		Query query = session.createQuery(hql);
+		query.setParameter("obsolete", true);
+		query.setParameter("ontology", ontology);
+		return (List<TermAlias>) query.list();
+	}
+
+	@SuppressWarnings({"unchecked"})
+	public List<String> getAllRelationships(Ontology ontology) {
+		Session session = HibernateUtil.currentSession();
+		String hql = " select distinct relationship.type from GenericTermRelationship as relationship";
+		if (ontology != null) {
+			hql += " where   relationship.termOne.ontology = :ontology OR relationship.termTwo.ontology = :ontology";
+		}
+		Query query = session.createQuery(hql);
+		if (ontology != null) {
+			query.setParameter("ontology", ontology);
+		}
+		return (List<String>) query.list();
+	}
+
+	/**
+	 * Retrieve all Relationships.
+	 *
+	 * @return list of relationships
+	 */
+	@SuppressWarnings({"unchecked"})
+	@Override
+	public List<String> getAllRelationships() {
+		Session session = HibernateUtil.currentSession();
+		String hql = " select distinct relationships.type from GenericTermRelationship as relationships";
+		Query query = session.createQuery(hql);
+		return (List<String>) query.list();
+	}
+
+	/**
+	 * Retrieve Term by OBO ID.
+	 *
+	 * @param termID term id
+	 * @return Generic Term
+	 */
+	@SuppressWarnings("unchecked")
+	public GenericTerm getTermByOboID(String termID) {
+
+		Session session = HibernateUtil.currentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<GenericTerm> cr = cb.createQuery(GenericTerm.class);
+		Root<GenericTerm> root = cr.from(GenericTerm.class);
+		cr.select(root).where(cb.equal(root.get("oboID"), termID));
```

Comment:
This syntax is hard for me to grok, but I guess I'll have to get used to it. Does the "setCacheable" call in the previous version do anything important that we need to preserve?